### ISSUE TITLE
Firefox Android - JSON export tool exports page source not custom fields

### ISSUE THREAD

**Issue by curious-toad** [AdvancedCustomFields/acf#425]
Experiencing an odd bug with the json export tool on Android version of Firefox.

Noticed the issue when trying to migrate new fields from my staging site to live site (both on same server). I'm getting the "Import file empty" error, even though the file is not blank. 

In inspecting the file there's no custom fields anywhere. Instead of exporting the fields as json, the tool is exporting the raw html for the current page. Sample output here:

Page: http://www.example.com/wp-admin/edit.php?post_type=acf-field-group&page=acf-tools

JSON file:

`<!DOCTYPE html>
<html class="wp-toolbar"
	lang="en-GB">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	<title>Tools &lsaquo; WaterBear &#8212; WordPress</title>
<script type="text/javascript">
addLoadEvent = function(func){if(typeof jQuery!=='undefined')jQuery(document).ready(func);else if(typeof wpOnload!=='function'){wpOnload=func;}else{var oldonload=wpOnload;wpOnload=function(){oldonload();func();}}};
var ajaxurl = '/wp-admin/admin-ajax.php',
	pagenow = 'custom-fields_page_acf-tools',
	typenow = 'acf-field-group',
	adminpage = 'custom-fields_page_acf-tools',
	thousandsSeparator = ',',
	decimalPoint = '.',
	isRtl = 0;
</script>
<link rel='dns-prefetch' href='//s.w.org' />
<style type="text/css">
img.wp-smiley,
img.emoji {
	display: inline !important;
	border: none !important;
	box-shadow: none !important;
	height: 1em !important;
	width: 1em !important;
	margin: 0 .07em !important;
	vertical-align: -0.1em !important;
	background: none !important;
	padding: 0 !important;
}
</style>
	<link rel='stylesheet' id='dashicons-css'  href='http://www.example.com/wp-includes/css/dashicons.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='admin-bar-css'  href='http://www.example.com/wp-includes/css/admin-bar.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='common-css'  href='http://www.example.com/wp-admin/css/common.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='forms-css'  href='http://www.example.com/wp-admin/css/forms.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='admin-menu-css'  href='http://www.example.com/wp-admin/css/admin-menu.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='dashboard-css'  href='http://www.example.com/wp-admin/css/dashboard.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='list-tables-css'  href='http://www.example.com/wp-admin/css/list-tables.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='edit-css'  href='http://www.example.com/wp-admin/css/edit.min.css?ver=5.6' media='all' />
<link rel='stylesheet' id='revisions-css'  href='http://www.example.com/wp-admin/css/revisions.min.css?ver=5.6' media='all' />

..... etc `

As you can see, that's all pretty standard html for the WordPress admin pages. 

Currently running WordPress 5.6. As this release is after the big changes around jQuery versions I thought I'd try an earlier version too, but the issue is present on 5.3.

The tool works fine using Firefox desktop (MacOS and Windows). Chrome Android works OK. Not sure about Safari on iOS - I sometimes get similar bugs to FF but I'm without a device to check on.

I'm using Chrome as a workaround for now but would be good to know what is going on with Firefox.

Thanks for your help (and in general for making such a great plugin).

Rob

**Comment by curious-toad** [comment]
I've also tested the Generate PHP tool for the same field groups and it works on Firefox Android, so this must be something to do with the actual downloading of the json file? Is there a way to output the json on page in the same way the Generate PHP tool functions?

**Comment by elliotcondon** [comment]
Hi @curious-toad 

Thanks for the bug report - how strange. Truth be told, we haven't performed much testing with the Export tool on mobile devices. It's possible this may be a limitation of the plugin, but I'll be happy to investigate this with the team in the new year.

**Comment by curious-toad** [comment]
Thanks Elliott, appreciate your time. Mobile use is a bit of an edge case - it's been useful for me to be able to sort small amends in the small hours whilst on paternity leave, otherwise I probably wouldn't have ever noticed it!

After digging about a bit more I think this couldbe related to Firefox itself - the Android version recently received a major upgrade and there are a lot of reported issues with handling downloads. For example: https://github.com/mozilla-mobile/fenix/issues/16289

So there is a chance that this will be resolved with a fix their end. 

Images and other media formats seem OK on the whole but json seems troublesome. This link is a very basic example, but I looked for an example json file where users have to download via a button, in the same manner as the ACF export tool. Towards the bottom of the page the buttons and  text links to download the json sample files work in Firefox Android: https://www.learningcontainer.com/sample-json-file/

So it looks like the browser isn't picking up the file header / information or something when trying to download in ACF.

Hope that helps - as I say, it's a relatively edge case and there is a workaround (use alternative browser for now).

**Comment by curious-toad** [comment]
Hi @elliotcondon - I've just had to export/import a field group on a new device (with latest version of Android and Firefox) and it was working a treat. I'm guessing this was actually an issue with Firefox moving to a new download manager as part of one of their big core updates.

Sorry for the red herring - doesn't look like it was to do with the ACF export specifically. I'll close the issue now.
